crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1406,14 +1407,56 @@ impl<'db> TypeInferenceBuilder<'db> {
         let ast::StmtAugAssign {
             range: _,
             target,
-            op: _,
+            op,
             value,
         } = assignment;
-        self.infer_expression(value);
-        self.infer_expression(target);
 
-        // TODO(dhruvmanila): Resolve the target type using the value type and the operator
-        Type::Todo
+        // Resolve the target type, assuming a load context.
+        let target_type = match &**target {
+            Expr::Name(target) => self.infer_name_load(target),
+            Expr::Attribute(target) => self.infer_attribute_load(target),
+            _ => self.infer_expression(target),
+        };
+        let value_type = self.infer_expression(value);
+
+        // TODO(charlie): Add remaining branches for different types of augmented assignments.
+        if let (Operator::Sub, Type::Instance(class)) = (*op, target_type) {
+            let class_member = class.class_member(self.db, "__isub__");
+            let call = class_member.call(self.db, &[value_type]);
+
+            return match call.return_ty_result(self.db, AnyNodeRef::StmtAugAssign(assignment), self)
+            {
+                Ok(t) => t,
+                Err(e) => {
+                    self.add_diagnostic(
+                        assignment.into(),
+                        "unsupported-operator",
+                        format_args!(
+                            "Operator `{op}` is unsupported for type `{}`",
+                            target_type.display(self.db),
```

Comment:
I think this diagnostic probably should clarify that it's specific to augmented assignment, and should include the `value_type` as well.
```suggestion
                            "Operator `{op}=` is unsupported for type `{}`",
                            target_type.display(self.db),
                            value_type.display(self.db),
```